src/pybind/mgr/dashboard/HACKING.rst

```diff
@@ -838,3 +838,70 @@ updates its progress:
           task = TaskManager.run("dummy/task", {}, self._dummy)
           return task.wait(5)  # wait for five seconds
 
+
+How to deal with asynchronous tasks in the front-end?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+All executing and most recently finished asynchronous tasks are displayed on the
+"Backgroud-Tasks" menu.
+
+The front-end developer should provide a description, success message and error
+messages for each task on ``TaskManagerMessageService.messages``.
+This messages can make use of the task metadata to provide more personalized messages.
+
+When submitting an asynchronous task, the developer should provide a callback
+that will be automatically triggered after the execution of that task.
+This can be done by using the ``TaskManagerService.subscribe``.
+
+Most of the times, all we want to do after a task completes the execution, is
+displaying a notification message based on the execution result. The
+``TaskManagerNotificationService.notify`` will use the messages from
+``TaskManagerMessageService`` to display a success / error message based on the
+execution result of a task.
+
+Usage example:
+
+.. code-block:: javascript
+
+  export class TaskManagerMessageService {
+
+    messages = {
+      // Messages for 'rbd/create' task
+      'rbd/create': new TaskManagerMessage(
+        // Description
+        (metadata) => `Create RBD '${metadata.pool_name}/${metadata.image_name}'`,
+        // Success message
+        (metadata) => `RBD '${metadata.pool_name}/${metadata.image_name}'
+                       have been created successfully`,
+        // Error messages
+        (metadata) => {
+          return {
+            '17': `Name '${metadata.pool_name}/${metadata.image_name}' is already
+                   in use.`
+          };
+        }
+      ),
+      // ...
+    };
+
+    // ...
+  }
+
+  export class RBDFormComponent {
+    // ...
+
+    submit() {
+      // ...
+      this.rbdService.create(request).then((resp) => {
+        // Subscribe the submitted task
+        this.taskManagerService.subscribe('rbd/create',
+          {'pool_name': request.pool_name, 'rbd_name': request.name},
+          // Callback that will be invoked after task is finished
+          (finishedTask: FinishedTask) => {
+            // Will display a notification message (success or error)
+            this.taskManagerNotificationService.notify(finishedTask, finishedTask.ret_value.success);
```

Comment:
This file needs to be updated to reflect the latest changes.

Reply:
Done